pkg/spanconfig/spanconfigsqltranslator/sqltranslator.go

```diff
@@ -521,30 +521,29 @@ func (s *SQLTranslator) findDescendantLeafIDsForDescriptor(
 		return nil, nil // we're excluding this descriptor; nothing to do here
 	}
 
-	switch desc.DescriptorType() {
-	case catalog.Type, catalog.Schema:
-		// There is nothing to do for {Type, Schema} descriptors as they are not
-		// part of the zone configuration hierarchy.
-		return nil, nil
-	case catalog.Table:
+	var db catalog.DatabaseDescriptor
+	switch t := desc.(type) {
+	case catalog.TableDescriptor:
 		// Tables are leaf objects in the zone configuration hierarchy, so simply
 		// return the ID.
 		return descpb.IDs{id}, nil
-	case catalog.Database:
-	// Fallthrough.
+	case catalog.DatabaseDescriptor:
+		db = t
 	default:
-		return nil, errors.AssertionFailedf("unknown descriptor type: %s", desc.DescriptorType())
+		// There is nothing to do for non-table-or-database descriptors as they are
+		// not part of the zone configuration hierarchy.
+		return nil, nil
 	}
 
 	// There's nothing for us to do if the descriptor is offline or has been
 	// dropped.
-	if desc.Offline() || desc.Dropped() {
+	if db.Offline() || db.Dropped() {
 		return nil, nil
 	}
 
 	// Expand the database descriptor to all the tables inside it and return their
 	// IDs.
-	tables, err := descsCol.GetAllTableDescriptorsInDatabase(ctx, txn, desc.GetID())
+	tables, err := descsCol.GetAllTableDescriptorsInDatabase(ctx, txn, db)
```

Comment:
The changes in this file are all due to this method taking a `catalog.DatabaseDescriptor` now.